fixed version checker

PickleNik opened the pull request:
Solved a Medium level LeetCode problem lol
https://leetcode.com/problems/compare-version-numbers/
![image](https://user-images.githubusercontent.com/31113245/159195931-ab2ee590-a6b5-4b05-be51-9e88009a9579.png)

sy-b reviewed (approved):
# 👍

cyrildtm commented on Extensions/combined/popup.js:
```diff
@@ -123,15 +123,35 @@ function initializeVersionNumber() {
+  // goes through version numbers from left to right from greatest to least significant
+  for (let i = 0; i < latestarr.length; i++) {
+    let latest = parseInt(latestarr[i]);
+    let current = parseInt(currentarr[i]);
```
may generate index out of bound error

PickleNik replied:
> may generate index out of bound error

We're not changing the amount of version numbers. It will never be comparing 2.1.0.3 against 3.1.1.1.1.whatever

cyrildtm replied:
Just saying; I've seen people scratching head months after initial commit to find the origin of a NPE, and turns out to be a simple assumption that something will never happen.

PickleNik replied:
Adding checks just to make the code bulletproof adds unnecessary overhead which can hinder the performance.
In the only case that the initial 4 numbers match, but the latest version has more numbers separated by dots, the loop will simply fail, the function will return the default value of false and the version checker won't show up. No big deal.

PickleNik replied:
Adding checks just to make the code bulletproof adds unnecessary overhead which can hinder the performance.
In the only case that the initial 4 numbers match, but the latest version has more numbers separated by dots, the loop will simply fail, the function will return the default value of false and the version checker won't show up. No big deal.

cyrildtm replied:
```
compareVersions('2.3.0.1', '2.3.0')
false
```

sy-b replied:
For the record

As long as this doesn't change, we are safe
![extension version cannot be more that 4 dot seperated integers](https://user-images.githubusercontent.com/94835959/159637121-6b427a66-bffa-48a0-8d19-8dea9571456f.PNG)


Contrary to this 👆, the max possible acceptable is `3999999999` instead of `65536`.
Let's hope that the version 4 integer limit doesn't increase.

![extension max version 3999999999 3999999999 3999999999 3999999999](https://user-images.githubusercontent.com/94835959/159638095-a789f911-1421-4498-8dbe-159ae7c8d35d.PNG)

cyrildtm commented on Extensions/combined/popup.js:
```diff
@@ -123,15 +123,35 @@ function initializeVersionNumber() {
+    if (latest > current) {
+      outdated = true;
+      break;
+    } else if (latest < current) {
```
unnecessary if branch

cyrildtm replied:
Then Line 139 works

cyrildtm commented:
Per v3 documentation it's 1-4 *variable length* number. So making 2.3.0
after 2.2.9.1 is allowed, and making 2.3.0.1 thereafter is also allowed. I
just don't see the point of forcing it 2.3.0.0.

On Wed, Mar 23, 2022, 1:36 AM sy-b ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Extensions/combined/popup.js
> <https://github.com/Anarios/return-youtube-dislike/pull/515#discussion_r832895879>
> :
>
> >          document.getElementById("ext-update").style.padding = ".25rem .5rem";
>
>        }
>
>      });
>
>    // .catch(console.error);
>
>  }
>
>
>
> +// returns whether current < latest
>
> +function compareVersions(latestStr, currentStr) {
>
> +  let latestarr = latestStr.split(".");
>
> +  let currentarr = currentStr.split(".");
>
> +  let outdated = false;
>
> +  // goes through version numbers from left to right from greatest to least significant
>
> +  for (let i = 0; i < latestarr.length; i++) {
>
> +    let latest = parseInt(latestarr[i]);
>
> +    let current = parseInt(currentarr[i]);
>
>
> For the record
>
> As long as this doesn't change, we are safe
> [image: extension version cannot be more that 4 dot seperated integers]
> <https://user-images.githubusercontent.com/94835959/159637121-6b427a66-bffa-48a0-8d19-8dea9571456f.PNG>
>
> Contrary to this 👆, the max possible acceptable is 3999999999 instead of
> 65536.
> Let's hope that the version 4 integer limit doesn't increase.
>
> [image: extension max version 3999999999 3999999999 3999999999 3999999999]
> <https://user-images.githubusercontent.com/94835959/159638095-a789f911-1421-4498-8dbe-159ae7c8d35d.PNG>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Anarios/return-youtube-dislike/pull/515#discussion_r832895879>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABNHDBLKY567PSCU4U4E53LVBK3YBANCNFSM5RGK67LA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

PickleNik commented:
> Per v3 documentation it's 1-4 *variable length* number. So making 2.3.0
> after 2.2.9.1 is allowed, and making 2.3.0.1 thereafter is also allowed. I
> just don't see the point of forcing it 2.3.0.0.
> 
> On Wed, Mar 23, 2022, 1:36 AM sy-b ***@***.***> wrote:
> 
> > ***@***.**** commented on this pull request.
> > ------------------------------
> >
> > In Extensions/combined/popup.js
> > <https://github.com/Anarios/return-youtube-dislike/pull/515#discussion_r832895879>
> > :
> >
> > >          document.getElementById("ext-update").style.padding = ".25rem .5rem";
> >
> >        }
> >
> >      });
> >
> >    // .catch(console.error);
> >
> >  }
> >
> >
> >
> > +// returns whether current < latest
> >
> > +function compareVersions(latestStr, currentStr) {
> >
> > +  let latestarr = latestStr.split(".");
> >
> > +  let currentarr = currentStr.split(".");
> >
> > +  let outdated = false;
> >
> > +  // goes through version numbers from left to right from greatest to least significant
> >
> > +  for (let i = 0; i < latestarr.length; i++) {
> >
> > +    let latest = parseInt(latestarr[i]);
> >
> > +    let current = parseInt(currentarr[i]);
> >
> >
> > For the record
> >
> > As long as this doesn't change, we are safe
> > [image: extension version cannot be more that 4 dot seperated integers]
> > <https://user-images.githubusercontent.com/94835959/159637121-6b427a66-bffa-48a0-8d19-8dea9571456f.PNG>
> >
> > Contrary to this 👆, the max possible acceptable is 3999999999 instead of
> > 65536.
> > Let's hope that the version 4 integer limit doesn't increase.
> >
> > [image: extension max version 3999999999 3999999999 3999999999 3999999999]
> > <https://user-images.githubusercontent.com/94835959/159638095-a789f911-1421-4498-8dbe-159ae7c8d35d.PNG>
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Anarios/return-youtube-dislike/pull/515#discussion_r832895879>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/ABNHDBLKY567PSCU4U4E53LVBK3YBANCNFSM5RGK67LA>
> > .
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
> 

If it's 2.3.0 the version checker will still work. If after that 2.3.0.1 comes out, the version checker will not show up, but it will show up for all the following ones. But afaik we're not changing the amount of numbers in the version.

PickleNik commented:
If it's 2.3.0 the version checker will still work. If after that 2.3.0.1 comes out, the version checker will not show up, but it will show up for all the following ones. But afaik we're not changing the amount of numbers in the version.

PickleNik commented:
tbh it would take less time fixing it than trying to justify not to

PickleNik commented:
thanks @cyrildtm for a thorough review 👍
